src/ripple/app/tx/impl/Escrow.cpp

```diff
@@ -438,6 +442,23 @@ EscrowFinish::doApply()
             return tecCRYPTOCONDITION_ERROR;
     }
 
+    auto const sled = ctx_.view().peek(keylet::account((*slep)[sfDestination]));
+    if (ctx_.view().rules().enabled(featureDepositAuth))
+    {
+        // NOTE: Escrow payments cannot be used to fund accounts
+        if (! sled)
```

Comment:
I don't think that it is possible for this situation to *ever* trigger. See: https://github.com/ripple/rippled/blob/3e5490ef6db9cbc3abdff04655d80952b288d097/src/ripple/app/tx/impl/Escrow.cpp#L229-L234.

I understand that this check was already present further down, and you just cloned it here for completeness, and maybe it's just better to actually have this here anyways out of an abundance of caution, but I'm forced to wonder whether we leave the check in `EscrowCreate` and remove this checking code from `EscrowFinish` leaving only `assert(sled)` immediately after the lookup, and the following invariant check (warning: I just typed this out here; it's hasn't been compile-tested):

```
/**
 * @brief Invariant: an escrow must not create a new account root.
 */
class NoFundingEscrow
{
    bool fundedAccount_ = false;

public:

    void
    visitEntry(
        uint256 const&,
        bool,
        std::shared_ptr<SLE const> const&,
        std::shared_ptr<SLE const> const&);

    bool
    finalize(STTx const&, TER, beast::Journal const&);
    
};

void
NoFundingEscrow::visitEntry(
    uint256 const&,
    bool isDelete,
    std::shared_ptr <SLE const> const& before,
    std::shared_ptr <SLE const> const& after)
{
    // If an account root didn't exist before and exists after the transaction
    // then the transaction funded an account:
    if (!before && after && after->getType() == ltACCOUNT_ROOT)
        fundedAccount_ = true;
}

bool
NoFundingEscrow::finalize(STTx const&tx, TER, beast::Journal const& j)
{
    if (!fundedAccount_)
        return true;

    auto const type = tx.getTxnType();

    if (type == ttESCROW_CREATE || type == ttESCROW_FINISH || type == ttESCROW_CANCEL)
    {
        JLOG(j.fatal()) << "Invariant failed: an escrow operation funded an account";
        return false;
    }

    return true;
}
```

Coincidentally, another invariant checker we could add would be to ensure that _if_ an account root _is_ created (regardless of the transaction) that its initial balance is at greater than or equal to the reserve. We don't allow partially funding a new account.